editor/src/components/canvas/canvas-strategies/reparent-strategy-helpers.ts

```diff
@@ -22,22 +27,88 @@ import { InteractionSession, StrategyState } from './interaction-state'
 import { ifAllowedToReparent } from './reparent-helpers'
 import { getReparentCommands } from './reparent-utils'
 import { getDragTargets } from './shared-absolute-move-strategy-helpers'
+import Utils from '../../../utils/utils'
+import { reverse } from '../../../core/shared/array-utils'
+
+interface ReorderElement {
+  distance: number
+  centerPoint: CanvasPoint
+  closestSibling: ElementPath
+  siblingIndex: number
+}
 
 export function getReorderIndex(
   metadata: ElementInstanceMetadataMap,
   siblings: Array<ElementPath>,
   point: CanvasVector,
+  existingElement: ElementPath | null,
 ): number {
-  const targetSiblingIdx = siblings.findIndex((sibling) => {
+  let rowOrColumn: 'row' | 'column' | null = null
+
+  const first = siblings[0]
+  if (first != null) {
+    const parentPath = EP.parentPath(first)
+    if (parentPath != null) {
+      const parentMetadata = MetadataUtils.findElementByElementPath(metadata, parentPath)
+      if (parentMetadata?.specialSizeMeasurements.layoutSystemForChildren != 'flex') {
+        throw new Error(`Element ${EP.toString(parentPath)} is not a flex container`)
+      }
+
+      const flexDirection = parentMetadata?.specialSizeMeasurements.flexDirection
+      if (flexDirection != null) {
+        if (flexDirection.includes('row')) {
+          rowOrColumn = 'row'
+        } else if (flexDirection.includes('col')) {
+          rowOrColumn = 'column'
+        }
+      }
+    }
+  }
+
+  let reorderResult: ReorderElement | null = null
+  let siblingIndex: number = 0
+
+  for (const sibling of siblings) {
     const frame = MetadataUtils.getFrameInCanvasCoords(sibling, metadata)
-    return (
+    if (
       frame != null &&
-      rectContainsPoint(frame, point) &&
       MetadataUtils.isParentYogaLayoutedContainerAndElementParticipatesInLayout(sibling, metadata)
-    )
-  })
+    ) {
+      const centerPoint = Utils.getRectCenter(frame)
+      const distance = Utils.distance(point, centerPoint)
```

Comment:
I believe this distance should only take into account the relevant dimension based on the value of `rowOrColumn`.

Reply:
That only works if there's only a single row or column, if you imagine 9 elements that wrap to 3 rows of 3 elements then just the horizontal distance will make it jump around between the 3 rows.

Reply:
Ah yeah ok that makes sense 👍